✨ Feat : Project 리스트 조회하기 기능 구현

seonghyun-1 opened the pull request:
## PR 체크사항
PR이 다음 사항을 만족하는지 확인해주세요.

<!-- 
체크하려면 괄호 안에 "x"를 입력하세요. 
각 규칙은 Convention 문서에 있습니다.
PR 제목에 쓰는 prefix는 다음과 같습니다.
🚀 Release
🐛 Fix
✨ Feat
📝 Doc
♻️ Refactor
🔧 Chore
⏪️ Revert
🧪 Test
🎉 Init
-->

- [x] 커밋 제목 규칙
- [x] 커밋 메시지 작성 가이드라인
- [x] 라벨, 담당자, 리뷰어 지정


## PR 타입
어떤 유형의 PR인지 체크해주세요.

<!-- 체크하려면 괄호 안에 "x"를 입력하세요. -->

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:


## PR 설명
filter 및 category를 RequestParam으로 받습니다.
이 기준에 따라서 Project 리스트 배열을 반환해줍니다.
filter는 3가지입니다.
- 최신 순
- 오래된 순
- 인기 순

## 작업사항
- Project 리스트 배열을 반환
- ProjectRepository에 filter별로 3가지 메소드 구현

seonghyun-1 committed:
[feat] Project 생성기능 추가수정 ( Valid 및 Exception 추가 )

- Valid 추가
- ValidException 추가

seonghyun-1 committed:
[feat] Project 생성기능 추가(수정)

- dto이름 수정 
 - ProjectRequestDto -> ProjectCreateRequestDto

seonghyun-1 committed:
[feat] 프로젝트 생성기능(수정)
- exception아래 mapper폴더가 들어가있던 오류 수정

seonghyun-1 committed:
[feat] Project 수정기능 추가

- Project 수정기능을 추가

seonghyun-1 committed:
[feat] 프로젝트 리스트 조회기능
- 프로젝트 조회 기준 3가지 구현
  1. 최신순
  2. 오래된순
  3. 인기순

seonghyun-1 committed:
[fix] feature-#12 픽스
- 수정 및 주석 삭제

hyunmuam commented on src/main/java/com/team1/TBBCloneCoding/project/entity/Project.java:
```diff
@@ -18,6 +18,9 @@ public class Project extends TimeStamp {
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long projectId;
 
+    @Column(nullable = false)
```
카테고리는 Enum으로 관리할 수 있음 좋을 것 같아요. 근데 이건 프런트랑도 맞춰야 되는 부분이라 의견 부탁드립니다.

seonghyun-1 replied:
프론트엔드 담당자분과 확인해보고 팀문서에 남기기로 결정되었습니다!

hyunmuam reviewed (commented):
카테고리, 정렬 순서를 Enum으로 관리해보는 방법에 대해 생각해봐도 될 것 같습니다.

seonghyun-1 committed:
[feat] 프로젝트 추가기능 (화)
- 중간저장

seonghyun-1 committed:
[feat] 프로젝트 추가기능 (화)
- 중간저장2

seonghyun-1 committed:
[feat] 프로젝트 요구사항 추가
- projectListResponseDto 생성 및 요구사항을 반영한 변수 추가
 - totalSupport, percent, goalPrice 추가 반환
- 좋아요 갯수 반환 추가
 - projectLike
